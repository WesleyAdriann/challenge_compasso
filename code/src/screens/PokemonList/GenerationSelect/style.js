import styled from 'styled-components/native'

import { COLORS, globalStyles } from '../../../style'

export const GenerationSelectWrapper = styled.View`
  margin: 0 8px 8px;
`

export const Label = styled.Text`
  ${globalStyles.defaultFont}
  color: ${COLORS.primaryBlack};
  font-size: 12px;
  margin-bottom: 2px;
  text-align: center;
`

export const OptionsContainer = styled.View`
  flex-direction: row;
  justify-content: space-between;
`
