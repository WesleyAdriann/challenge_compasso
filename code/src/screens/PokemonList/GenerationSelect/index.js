import React from 'react'
import PropTypes from 'prop-types'

import { GenerationSelectWrapper, Label, OptionsContainer, GenerationButton } from './style'

const GenerationSelect = ({ generations, selected, handleSelectGeneration }) => (
  <GenerationSelectWrapper>
    <Label>Generations</Label>
    <OptionsContainer>
      {
        generations.map((generation, index) => (
          <GenerationButton
            key={index}
            selected={selected === generation.id}
            onPress={() => handleSelectGeneration(generation.id)}
          >
            {
              selected === generation.id
                ? <GenerationButton.Icon />
                : <GenerationButton.Text>{generation.id}</GenerationButton.Text>
            }
          </GenerationButton>
        ))
      }
    </OptionsContainer>
  </GenerationSelectWrapper>
)

GenerationSelect.propTypes = {
  generations: PropTypes.arrayOf(PropTypes.object).isRequired,
  selected: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  handleSelectGeneration: PropTypes.func
}

GenerationSelect.defaultProps = {
  handleSelectGeneration: () => {},
  selected: 0
}

export default GenerationSelect
