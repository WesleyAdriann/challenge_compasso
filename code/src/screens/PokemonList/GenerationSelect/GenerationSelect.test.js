import React from 'react'
import renderer from 'react-test-renderer'

import GenerationSelect from './index'

describe('GenerationSelect Component', () => {
  test('renders correctly', () => {
    const props = {
      selected: 0,
      handleSelectGeneration: jest.fn(),
      generations: [],
    }

    const tree = renderer.create(<GenerationSelect {...props}/>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
