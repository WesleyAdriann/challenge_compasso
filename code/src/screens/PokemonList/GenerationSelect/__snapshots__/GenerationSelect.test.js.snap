// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenerationSelect Component renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "marginBottom": 8,
        "marginLeft": 8,
        "marginRight": 8,
        "marginTop": 0,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#050505",
          "fontFamily": "Roboto",
          "fontSize": 12,
          "marginBottom": 2,
          "textAlign": "center",
        },
      ]
    }
  >
    Generations
  </Text>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "space-between",
          "paddingRight": 8,
        },
      ]
    }
  />
</View>
`;
